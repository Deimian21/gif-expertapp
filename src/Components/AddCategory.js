import React, { useState } from 'react'
import Proptypes from 'prop-types';

export const AddCategory = ({setCategory}) => {
    const [imputValiu, setimputValiu] = useState('');
    const handleInputChange= (e) =>{
     setimputValiu(e.target.value);
    }
    const handleSumbit = (e) =>{
      e.preventDefault();
      if (imputValiu.trimEnd().length>2) {
        setCategory( cats => [imputValiu,...cats]);
        setimputValiu('');
      }
      
    }
    return (
        <form onSubmit={handleSumbit}>
            
            <h2> AÃ±adir componete</h2>
            <input type='text' value={imputValiu} onChange={handleInputChange}></input>
            </form>
    )
}

AddCategory.propTypes ={
  setCategory:Proptypes.func.isRequired
}
