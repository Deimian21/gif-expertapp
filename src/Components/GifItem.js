import React from 'react'
import PropTypes from 'prop-types';
export const GifItem = ({url,title}) => {
    //console.log(props);
    return (
        <div className='animate__fadeIn'>
           <img src={url} alt={title}/>
           <p>{title}</p>
        </div>
    )
}
GifItem.propTypes= {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
    }