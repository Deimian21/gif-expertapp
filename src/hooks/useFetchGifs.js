import { useEffect, useState } from "react"
import { getGif } from '../helpers/getGifts';

export const useFetchGifs = (category) => {
  const [state, setstate] = useState({
      data:[],
      loading: true
    
    });

        useEffect(()=>{
        getGif(category)
        .then(imgs => { 
                setstate({
                    data:imgs,
                    loading: false
                });
        })
    },[category]  )

   /*  setTimeout(() => {
        setstate({
            data:[1,2,3,4,5,6,7],
            loading: false
        })
    },3000); */

    return state;
}
