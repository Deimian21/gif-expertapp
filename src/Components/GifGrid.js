//import { logDOM } from '@testing-library/react';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
     const {data:images,loading} = useFetchGifs(category);

    return (
        <>
        <h3 className='animate__fadeIn'>{category}</h3>
        { loading && <p>Loading</p>}
        <div className='card-grid'>
          
            { 

            images.map( img =>(
            <GifItem 
            key={img.id}
            {...img}/>
            ) )

            }
        </div>
        </>
    )
}
GifGrid.propTypes = {
    category:PropTypes.string.isRequired
}
