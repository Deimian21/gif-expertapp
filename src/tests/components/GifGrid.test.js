import { shallow } from 'enzyme';
import React from 'react';
import{GifGrid} from '../../Components/GifGrid';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('Pruebas en GifGrid', () => {
    const category='Dragon Ball';
   
 test('debe tener un snapshot', () => {
     useFetchGifs.mockReturnValue({
         data:[],
         loading:true
     });
    const wrapper= shallow(<GifGrid category={ category }/>);
    expect(wrapper).toMatchSnapshot();
 });
     
    test('debe de mostrar items cuando se cargan imagenes en use fetch git  ', () => {
        const gifs=[{
            id:'ABC',
            url:'https://localhost/imagenes/imagen.png',
            title:'una cosa'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        const wrapper= shallow(<GifGrid category={ category }/>);
        expect(wrapper.find('p').exists()).toBe(false);
        //expect(wrapper.find('GifGrid').length).toBe(gifs.length);
    })

})
