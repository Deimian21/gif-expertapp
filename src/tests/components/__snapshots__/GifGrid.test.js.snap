// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe tener un snapshot 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
